smtlib-backends-z3/tests/Main.hs

```diff
@@ -17,7 +18,7 @@ main = do
         testBackend "Error handling" failingSources z3
       ]
   where
-    z3 todo = Z3.with $ todo . Z3.toBackend
+    z3 todo = Z3.with def $ todo . Z3.toBackend
```

Comment:
But that's not possible, is it ? Since `def` is imported from `Data.Default`. Unless there's a way to do something like
```haskell
import Data.Default (def as defaultConfig)
```
?

Reply:
Nevermind I can just do `let defaultConfig = def in ...`.

Reply:
Because the need to use `defaultConfig` would be recurrent, I'd define in the Z3 backend
```
defaultConfig :: Config
defaultConfig = Config []

instance Default Config where
  def = defaultConfig
```
I would also drop the `Default Config` instance, but I'm not sure what you want it for.

Reply:
I guess the point is that it doesn't hurt to have it and it makes it easier to get started since you don't have to think about an actual configuration. So just like for the `Process` backend, although it is indeed not as useful here.